table_engine/src/engine.rs

```diff
@@ -315,6 +313,9 @@ pub trait TableEngine: Send + Sync {
     /// Close the engine gracefully.
     async fn close(&self) -> Result<()>;
 
+    /// Validate the request of create table.
+    async fn validate_create_table(&self, request: &CreateTableParams) -> Result<()>;
```

Comment:
Add a comment to explain why this method is needed? I think it seems a strange acutally...